mpisppy/generic_cylinders.py

```diff
@@ -493,6 +496,11 @@ def _do_EF(module, cfg, scenario_creator, scenario_creator_kwargs, scenario_deno
         sputils.ef_ROOT_nonants_npy_serializer(ef, f'{cfg.solution_base_name}.npy')
         sputils.write_ef_tree_solution(ef,f'{cfg.solution_base_name}_soldir')
         global_toc("Wrote EF solution data.")
+        # callout to special solution writer if it exists
+        if hasattr(module, 'custom_writer'):
+            module.custom_writer(ef, cfg)    
```

Comment:
```suggestion
    # callout to special solution writer if it exists
    if hasattr(module, 'custom_writer'):
        module.custom_writer(ef, cfg)    
```
Placing it outside the if block, so that it is called independently of the value of `cfg.solution_base_name`, like for `_do_decomp()`